module/VuFind/src/VuFind/ILS/Driver/XCNCIP2.php

```diff
@@ -173,6 +193,11 @@ public function init()
         }
         $this->disableRenewals
             = $this->config['Catalog']['disableRenewals'] ?? false;
+
+        $this->useOAuth2 = ($this->config['tokenEndpoint'] ?? false)
+            && ($this->config['clientId'] ?? false)
+            && ($this->config['clientSecret'] ?? false);
+        $this->config['tokenBasicAuth'] = $this->config['tokenBasicAuth'] ?? false;
```

Comment:
Perhaps make `$tokenBasicAuth` a member variable like `$useOAuth2`, so you don't need to modify $config.

Reply:
done